d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -610,7 +638,19 @@ public void processResult(int rc, String path, Object ctx, Stat stat)
     protected void processWatch(String propertyName, WatchedEvent event)
     {
       // Reset the watch
-      _zk.getChildren(getPath(propertyName), this, this, false);
+      if (_readWindowMs > 0 && _executorService != null)
+      {
+        // Delay setting the watch based on configured _readWindowMs
+        int midPoint = _readWindowMs / 2;
+        int delay = midPoint + ThreadLocalRandom.current().nextInt(midPoint);
```

Comment:
So the delay is in the range of [midPoint, readWindowMs] -- why not [0, readWindowMs]? There is no need to explicitly add latency for the calls.

Reply:
The purpose of adding the delay is to reduce the number of reads as discussed in the design document. This will help us to reduce a lot of unwanted watch updates by delaying within the acceptable threshold. Please check the design document and let me know you still have concerns with this approach.

Reply:
Thanks for the clarification. Reviewed the doc and agree there should be a delay.